msprime/simulations.py

```diff
@@ -819,57 +825,77 @@ class RecombinationMap(object):
+        values. To simulate a fixed number of loci, set this parameter to
+        ``True`` and scale ``positions`` to span the desired number of loci.
     """
-    DEFAULT_NUM_LOCI = 2**32 - 1
-    """
-    The default number of non-recombining loci in a RecombinationMap.
-    """
-    def __init__(self, positions, rates, num_loci=None):
-        if num_loci is None:
-            num_loci = self.DEFAULT_NUM_LOCI
+    def __init__(self, positions, rates, num_loci=None, discrete=False):
+        if num_loci is not None:
+            if num_loci == positions[-1]:
+                warnings.warn(
+                        "num_loci is deprecated and should not be used. "
+                        "The default behavior is that the number of loci "
+                        "is equal to the length of the sequence.",
+                        DeprecationWarning)
+            else:
+                raise ValueError(
+                        "num_loci does not match sequence length. "
+                        "To set a discrete number of recombination sites, "
+                        "scale positions to span the desired number of loci "
+                        "and set discrete=True")
         self._ll_recombination_map = _msprime.RecombinationMap(
-            num_loci, positions, rates)
+            positions, rates, discrete)
 
     @classmethod
-    def uniform_map(cls, length, rate, num_loci=None):
+    def uniform_map(cls, length, rate, num_loci=None, discrete=False):
         """
         Returns a :class:`.RecombinationMap` instance in which the recombination
         rate is constant over a chromosome of the specified length. The optional
-        ``num_loci`` controls the number of discrete loci in the underlying
-        simulation, and is by default large enough to be effectively be
-        a continuous model.
+        ``discrete`` controls whether recombination sites can occur only on integer
+        positions or can take continuous values. The optional ``num_loci`` is
+        deprecated and should not be used.
 
         The following map can be used to simulate a true finite locus model
         with a fixed number of loci ``m``::
 
-            >>> recomb_map = RecombinationMap.uniform_map(m, rate, num_loci=m)
+            >>> recomb_map = RecombinationMap.uniform_map(m, rate, discrete=True)
+
+        .. warning::
+            The ``num_loci`` parameter is deprecated. To set a discrete number of
+            possible recombination sites along the sequence, scale ``positions``
+            to the desired number of sites and set ``discrete=True`` to ensure
+            recombination occurs only at integer values.
 
         :param float length: The length of the chromosome.
         :param float rate: The rate of recombination per unit of sequence length
             along this chromosome.
-        :param int num_loci: The number of discrete loci in the underlying
+        :param int num_loci: **This parameter is deprecated**.
```

Comment:
this parameter is no longer supported

Reply:
Delete the old description -- we just want to tell the user that this positional parameter is there, and that it shouldn't be used.